test/run_clippy.sh

```diff
@@ -0,0 +1,138 @@
+#!/bin/bash
+
+set -eoux pipefail
+
+sudo hab pkg install core/bzip2
```

Comment:
We are running as `buildkite-agent` but hab is installed as root.

Reply:
*Must* hab be installed as root?

Reply:
If we install it the (as far as I know) recommended way, we install it as root. I'm curious about how you'd otherwise install it. the curl|bash method installs as root (obviously, with the sudo bash bit)

Reply:
"it" being habitat? There's no need to install it as root unless you need to run stuff as root. If all we're doing is installing some packages to allow `cargo clippy` to work, I think we might be able to get away with not being root. It's fine for now, but an issue to see if we can avoid being root here would be nice to file for follow-up.